### DISCUSSION TITLE
[Quick question]How to select a group cells in a mesh?

### DISCUSSION THREAD

**Discussion by ZiguoAtGitHub** [marcomusy/vedo#748] [Q&A] [answered]
Hi Marcomusy and other masters
I create a mesh with Mesh([verts,cells]], which includes a large number ploygons. How can I select a group of ploygons and hide them?
Thank you all!

**Comment by marcomusy** [comment]
Hi,
What do you mean by "select"? You need to be more specific.

**Reply by ZiguoAtGitHub** [reply]
I look in vedo's doc. maybe the meaning of "select" is same as "extract" ,:)

**Comment by ZiguoAtGitHub** [comment, marked as answer]
Dear Marco,@marcomusy
I should explain my problem more clearly with the following codes:

```
"""Manually build a mesh from points and faces"""
from vedo import Mesh, show

verts = [(50,50,50), (70,40,50), (50,40,80), (80,70,50)]
faces = [(0,1,2), (2,1,3), (1,0,3)]
''' (the first triangle face is formed by vertex 0, 1 and 2)
Build the polygonal Mesh object:'''
mesh = Mesh([verts, faces])
mesh.backcolor('violet').linecolor('tomato').linewidth(2)
labs = mesh.labels('id').c('black')
#-----------------------------------------------------
```
**My problem is: How to show the first triangle only, but keep the codes above the dashed line UNCHANGED**

`show(mesh, labs, __doc__, viewup='z', axes=1).close()`

Please let me know if you can not get my point. Thank you again.

BTW: 
If I create each mesh for every single triangle, I can specify the first mesh on and off for the other two.
But for a large number triangles, the speed for plot will be decreased dramatically. I already tried.

**Reply by marcomusy** [reply]
Is this solution working for you?
The fourth number is the opacity which can be set to 0 or to 255.
The first 3 are RGB values.
 
```python

from vedo import Mesh, show

verts = [(50,50,50), (70,40,50), (50,40,80), (80,70,50)]
faces = [(0,1,2), (2,1,3), (1,0,3)]

mesh = Mesh([verts, faces])
#-----------------------------------------------------

mesh.cell_individual_colors([(255,0,0,255),(0,255,0,20),(0,0,120,20)])

show(mesh, axes=1)
```
<img width="394" alt="Screenshot 2022-12-08 at 15 16 33" src="https://user-images.githubusercontent.com/32848391/206469107-f9731e85-9f66-4d5b-8605-cde21060dc89.png">

**Reply by marcomusy** [reply]
PS: note that if you have defined some color mapping you can retrieve the current RGBA values and mask the faces you like by setting to zero the fourth column, eg:

```python

from vedo import Mesh, show

verts = [(50,50,50), (70,40,50), (50,40,80), (80,70,50)]
faces = [(0,1,2), (2,1,3), (1,0,3)]

mesh = Mesh([verts, faces])
mesh.cmap("jet", mesh.cell_centers()[:,2], on="cells")
#-----------------------------------------------------

print(mesh.retrieve_rgb(on="cells", alpha=True))

# mesh.cell_individual_colors([...])

show(mesh, axes=1)
```

**Reply by ZiguoAtGitHub** [reply]
Hi Marco,
First of all, thank you for your kindly reply of every question!

I am not good at asking questions, especially on web. Sorry about that.

In fact, I want to control every individual cell in a mesh, its color, its visibility and etc.
The lazy way I had tried is creating independent mesh for every triangle. It works well for very smaller numbers of triangles. When the number of triangles increases to over 10k, the plotting speed will be decreased severely as I noted previously.

Another doable method is calling plt.clear(), then creating new meshes and plot them every time. It also works.
But there is one little unpleasant problem: when I change to plot new meshes the plot window will be blank for several seconds.
Is this the best way? Would you like to give some hints for me?

I have no idea about whether vedo can provides more flexibility on individual cell in a mesh or not.
If not, I am very looking forward for that.

Thank you again!
🙂

**Reply by ZiguoAtGitHub** [reply]
Dear Marco,
Please forget the above question about the second method. I improve the codes the problem is gone.
🙂

**Reply by marcomusy** [reply]
Note that  function will change name in the next release from
`mesh.retrieve_rgb(on="cells", alpha=True)`
to
`mesh.get_rgba("cells")`

**Reply by ZiguoAtGitHub** [reply]
OK. Thanks.

**Comment by marcomusy** [comment]
Hi, no worries!
why is the above solution not viable in your case?
